core/domain/auth_services.py

```diff
@@ -282,3 +288,50 @@ def revoke_super_admin_privileges(user_id: str) -> None:
         user_id: str. The Oppia user ID to revoke privileges from.
     """
     firebase_auth_services.revoke_super_admin_privileges(user_id)
+
+
+def get_csrf_secret_value() -> str:
+    """Returns the csrf secret value if it exists otherwise initializes it.
+
+    Returns:
+        str. Returns the csrf secret value.
+    """
+    memcached_items = caching_services.get_multi(
+        caching_services.CACHE_NAMESPACE_DEFAULT,
+        None,
+        [CSRF_SECRET_INSTANCE_ID]
+    )
+    if CSRF_SECRET_INSTANCE_ID in memcached_items:
+        csrf_value = memcached_items[CSRF_SECRET_INSTANCE_ID]
+        # Rulling out the possibility for csrf_value of being type other
+        # than str in order to avoid mypy error.
+        assert isinstance(csrf_value, str)
+        return csrf_value
+
+    csrf_secret_model = auth_models.CsrfSecretModel.get(
+        CSRF_SECRET_INSTANCE_ID, strict=False)
+
+    if csrf_secret_model is None:
+        csrf_secret_value = base64.urlsafe_b64encode(os.urandom(20)).decode()
+        auth_models.CsrfSecretModel(
+            id=CSRF_SECRET_INSTANCE_ID,
+            oppia_csrf_secret=csrf_secret_value
+        ).put()
+        caching_services.set_multi(
+            caching_services.CACHE_NAMESPACE_DEFAULT,
+            None,
+            {
+                CSRF_SECRET_INSTANCE_ID: csrf_secret_value
+            }
+        )
+
+    csrf_secret_model = auth_models.CsrfSecretModel.get(
```

Comment:
You fetched this on line 311. Why fetch it again here? Should this be within the "if" block?

Reply:
Yes, that makes sense. This needs to be in the "if" block, this will help us reduce the calls to the model.
Thanks, I have updated this.